sw/device/sca/aes_serial.c

```diff
@@ -66,7 +83,15 @@ static void aes_serial_set_key(const uint8_t *key, size_t key_len) {
  * @return Result of the operation.
  */
 static void aes_serial_encrypt(const uint8_t *plaintext, size_t plaintext_len) {
-  aes_data_put_wait(plaintext);
+  bool ready = false;
+  do {
+    SS_CHECK(dif_aes_get_status(&aes, kDifAesStatusInputReady, &ready) ==
+             kDifAesOk);
+  } while (!ready);
+  dif_aes_data_t data;
+  SS_CHECK(plaintext_len <= sizeof(data.data));
+  memcpy(data.data, plaintext, plaintext_len);
```

Comment:
It would be nice if we could avoid the `memcpy` here. But it would require you to change the function interface.